File: src/java/util/Map.ts
class java.util.Map {
// Methods
public abstract clear(): void {
this.clear()
}
public compute(obj: Object, bif: BiFunction): Object {
return this.compute(obj, bif)
}
public computeIfAbsent(obj: Object, fun: Function): Object {
return this.computeIfAbsent(obj, fun)
}
public computeIfPresent(obj: Object, bif: BiFunction): Object {
return this.computeIfPresent(obj, bif)
}
public abstract containsKey(obj: Object): boolean {
return this.containsKey(obj)
}
public abstract containsValue(obj: Object): boolean {
return this.containsValue(obj)
}
public abstract entrySet(): Set {
return this.entrySet()
}
public abstract equals(obj: Object): boolean {
return this.equals(obj)
}
public forEach(bic: BiConsumer): void {
this.forEach(bic)
}
public abstract get(obj: Object): Object {
return this.get(obj)
}
public getOrDefault(obj: Object, obj_1: Object): Object {
return this.getOrDefault(obj, obj_1)
}
public abstract hashCode(): number {
return this.hashCode()
}
public abstract isEmpty(): boolean {
return this.isEmpty()
}
public abstract keySet(): Set {
return this.keySet()
}
public merge(obj: Object, obj_1: Object, bif: BiFunction): Object {
return this.merge(obj, obj_1, bif)
}
public abstract put(obj: Object, obj_1: Object): Object {
return this.put(obj, obj_1)
}
public abstract putAll(map: Map): void {
this.putAll(map)
}
public putIfAbsent(obj: Object, obj_1: Object): Object {
return this.putIfAbsent(obj, obj_1)
}
public abstract remove(obj: Object): Object {
return this.remove(obj)
}
public remove(obj: Object, obj_1: Object): boolean {
return this.remove(obj, obj_1)
}
public replace(obj: Object, obj_1: Object): Object {
return this.replace(obj, obj_1)
}
public replace(obj: Object, obj_1: Object, obj_2: Object): boolean {
return this.replace(obj, obj_1, obj_2)
}
public replaceAll(bif: BiFunction): void {
this.replaceAll(bif)
}
public abstract size(): number {
return this.size()
}
public abstract values(): Collection {
return this.values()
}
// Fields
}

File: src/java/util/Map.java
public abstract interface class java.util.Map
{
    // Constructors
    // Methods
    public abstract void clear();
    public java.lang.Object compute(java.lang.Object, java.util.function.BiFunction);
    public java.lang.Object computeIfAbsent(java.lang.Object, java.util.function.Function);
    public java.lang.Object computeIfPresent(java.lang.Object, java.util.function.BiFunction);
    public abstract boolean containsKey(java.lang.Object);
    public abstract boolean containsValue(java.lang.Object);
    public abstract java.util.Set entrySet();
    public abstract boolean equals(java.lang.Object);
    public void forEach(java.util.function.BiConsumer);
    public abstract java.lang.Object get(java.lang.Object);
    public java.lang.Object getOrDefault(java.lang.Object, java.lang.Object);
    public abstract int hashCode();
    public abstract boolean isEmpty();
    public abstract java.util.Set keySet();
    public java.lang.Object merge(java.lang.Object, java.lang.Object, java.util.function.BiFunction);
    public abstract java.lang.Object put(java.lang.Object, java.lang.Object);
    public abstract void putAll(java.util.Map);
    public java.lang.Object putIfAbsent(java.lang.Object, java.lang.Object);
    public abstract java.lang.Object remove(java.lang.Object);
    public boolean remove(java.lang.Object, java.lang.Object);
    public java.lang.Object replace(java.lang.Object, java.lang.Object);
    public boolean replace(java.lang.Object, java.lang.Object, java.lang.Object);
    public void replaceAll(java.util.function.BiFunction);
    public abstract int size();
    public abstract java.util.Collection values();
    // Fields
}